src/test/kotlin/RankTest.kt

```diff
@@ -0,0 +1,30 @@
+package lotto
+
+import org.junit.jupiter.api.Test
+import kotlin.test.DefaultAsserter.assertEquals
+
+class RankTest {
+    @Test
+    fun `should assign the corresponding rank when it has bonus`() {
+        val matchingNumber = 5
+        val hasBonus = true
+
+        assertEquals("Should match the right rank", Rank.valueOf(matchingNumber, hasBonus).toString(), "SECOND")
```

Comment:
```suggestion
        assertEquals("Should match the right rank", Rank.valueOf(matchingNumber, hasBonus), Rank.SECOND)
```
Since we're using `assertEquals`, we can directly compare enum values to check if we're getting the expected result.

Why does this work?
What's special about `enum` classes in Kotlin (and Java), and what does it actually mean when two enum values are considered `"equal"`?

Reply:
In my research enum classes are objects, so comparing it as objects (their actually location on the memory) is better (memory-safe) than comparing two strings. Enum classes are special because they hold a fixed set of constant values which are all a singleton object. In Kotlin we can define properties and functions for each enum value, and also override  or create custom factory methods. Thanks for made me research those concepts again! 😄